### ISSUE TITLE
Some Questions about MACE for pure water and in general.

### ISSUE THREAD

**Issue by nihil39** [ACEsuit/mace#576]
Hi, 

I'm trying to train a model for pure water made with MACE. My dataset consists of 3000 configuration at 300K taken at different densities, from 0.972 to 1.04 $\frac{g}{cm^3}$ coming from DFT calculations. 

I have some questions, sorry if they are trivial:

I am trying to use the "right" hyperparameters. In order to do so: I am starting reading [this](https://mace-docs.readthedocs.io/en/latest/examples/training_examples.html#liquid-water) and comparing those parameters with the ones written [here](https://github.com/ACEsuit/mace-mp/blob/main/mace_mp_0/2023-12-03-mace-128-L1.sh) that should be the ones used for the MACE MP-0 L1 model. 

1. What are the advised parameters for pure water? In particular the Energy and the forces weight. 

2. What do the following hyperparameters mean?

```
--interaction_first="RealAgnosticResidualInteractionBlock" \
    --interaction="RealAgnosticResidualInteractionBlock" \
    --num_interactions=2 \
    --correlation=3 \

``` 
3. Is this the max order of the spherical harmonics ``` -max_ell=3 \``` used in the model? Why it is the same for the L0, L1 and L2  MACE MP0 models? 

4. 
What does "last readout" mean? What is the width of radial MLP?
```
--MLP_irreps MLP_IRREPS
   hidden irreps of the MLP in last readout
 
--radial_MLP RADIAL_MLP
  width of the radial MLP
```

 
5. Last and maybe least important. How can I properly set a wandb interface? Where should I put my wandb token? In --wandb_entity? What are the possible loggable hyperparameters, how can I select them?

Thank you in advance.

**Comment by gabor1** [comment]
MAke sure you use the latest version in the main branch. The code is evolving fast. I suggest you use "1" and "1" for energies and forces weights. use `--model ="MACE"` and don't specify any interaction blocks. Both the parameter lists that you linked use average E0s (reference energies for atoms), we don't recommend this for general use, and indeed the latest version of the mace-mp-0 model doesn't do this. Compute isolated atom energies for H and O, and either put those into your dataset, or give the isolated atom energies as E0s using command line options. 

-max_ell is the spherical harmonics order for the local environment, use 3. L is the spherical harmonic order of the messages in the message passing step, 0 is invariant, 1 and 2 are higher order (equivariant). Higher L will give you more accurate, but larger and slower models. 

"last readout" means the readout function of the last layer. We typically put a shallow MLP on it. the radial MLP is the parametrisation of the radial basis used to expand the local environment. These are all defined and discussed in the MACE papers. 

I'll let others who use it regularly comment on wandb

**Comment by ilyes319** [comment]
I'll turn that into a discussion. Issues are for bugs and feature requests.
As a general comment, I suggest you use the help function of the arg parser, as it will provide some answers to these questions.

- For the loss weights, I would try forces: 10, energy: 1 for the go to. I always start with that.
- You should provide these hypers but for reference --correlation=3 says that MACE will construct 4 body messages at each layer. --num_interactions=2 means that it will have 2 layers. The type of interaction blocks is advanced and you should not specify them.
- For wandb, I suggest you read the wandb documentation. You should first login in your wandb account on your head node or computed (see https://docs.wandb.ai/ref/cli/wandb-login). Then you should select --wandb in the input script and and select the right project with --wandb_project. To select hypers to log, use --wandb_log_hypers. To see the list of what can use, use the help function of the arg parser.
